source/MaterialXContrib/MaterialXNode/MaterialXSurfaceOverride.cpp

```diff
@@ -0,0 +1,282 @@
+        // bool requiresLighting = (shaderRef != nullptr);
+        std::cout << "MaterialXSurfaceOverride: Create XML wrapper" << std::endl;
+        _glslWrapper->createWrapper(element);
+        // Get the fragment name
+        _fragmentName.set(_glslWrapper->getFragmentName().c_str());
+
+        // Register fragments with the manager if needed
+        //
+        MHWRender::MRenderer* theRenderer = MHWRender::MRenderer::theRenderer();
+        if (theRenderer)
+        {
+            MHWRender::MFragmentManager* fragmentMgr =
+                theRenderer->getFragmentManager();
+            if (fragmentMgr)
+            {
+                bool fragmentExists = (_fragmentName.length() > 0) && fragmentMgr->hasFragment(_fragmentName);
+                if (!fragmentExists)
+                {
+                    std::stringstream glslStream;
+                    _glslWrapper->getDocument(glslStream);
+                    std::string xmlFileName(Plugin::instance().getResourcePath().asString() + "/standard_surface_default.xml");
+
+                    // TODO: This should not be hard-coded
+                    fragmentMgr->setEffectOutputDirectory("d:/work/");
+                    fragmentMgr->setIntermediateGraphOutputDirectory("d:/work/");
+                    _fragmentName = fragmentMgr->addShadeFragmentFromFile(xmlFileName.c_str(), false);
+                }
+            }
+            std::cout << "MaterialXSurfaceOverride: Add XML fragment to manager: " << _fragmentName << std::endl;
+        }
+    }
+    catch (MaterialX::Exception& e)
+    {
+        std::cerr << "MaterialXSurfaceOverride: Failed to generate XML wrapper: " << e.what() << std::endl;
+    }
+
+    // TODO: Use our own image loader vs Mayas.
+    // MaterialX::StbImageLoaderPtr stbLoader = MaterialX::StbImageLoader::create();
+}
+
+MaterialXSurfaceOverride::~MaterialXSurfaceOverride()
+{
+	// TODO: Free sampler state here!
+
+	if (_glslWrapper)
+	{
+		delete _glslWrapper;
+	}
+}
+
+MHWRender::DrawAPI MaterialXSurfaceOverride::supportedDrawAPIs() const
+{
+	return MHWRender::kOpenGL | MHWRender::kOpenGLCoreProfile;
+}
+
+MString MaterialXSurfaceOverride::fragmentName() const
+{
+	return _fragmentName;
+}
+
+#if 0
+void MaterialXSurfaceOverride::getCustomMappings(MHWRender::MAttributeParameterMappingList& mappings)
+{
+	const MaterialX::StringMap& inputs = _glslWrapper->getPathInputMap();
+	for (auto i : inputs)
+	{
+		std::cout.rdbuf(std::cerr.rdbuf());
+		std::cout << "MaterialXSurfaceOverride: Get custom mappings: " << i.second.c_str() << std::endl;
+		MHWRender::MAttributeParameterMapping mapping(i.second.c_str(), "", false, true);
+		mappings.append(mapping);
+	}
+}
+#endif
+
+void MaterialXSurfaceOverride::updateDG()
+{
+}
+
+void MaterialXSurfaceOverride::updateShader(MHWRender::MShaderInstance& shader,
+                                            const MHWRender::MAttributeParameterMappingList& mappings)
+{
+    MStringArray params;

```

Comment:
As mentioned below. We can have reuse if we move the textureoverride code to MaterialXData.